var/spack/repos/builtin/packages/sundials/package.py

```diff
@@ -744,119 +742,95 @@ def libs(self):
-            tty.msg("Skipping sundials test: examples were not installed")
-            return
+        self.run_example(join_path("nvector", "cuda", "test_nvector_cuda"), ["10", "0", "0"], True)
 
-        for smoke_test in self._smoke_tests:
-            work_dir = join_path(self._smoke_tests_path, os.path.dirname(smoke_test[0]))
-            with working_dir(work_dir):
-                if smoke_test[3]:  # use cmake
-                    self.run_test(exe=cmake_bin, options=["."])
-                self.run_test(exe="make")
+    def test_cvadvdiff_cuda(self):
+        """build and run CUDA cvAdvDiff_kry"""
+        if "+cuda" not in self.spec or "+CVODE" not in self.spec:
+            raise SkipTest("Package must be installed with +cuda+CVODE")
 
-    def run_smoke_tests(self):
-        if "~examples-install" in self.spec:
-            return
-
-        for smoke_test in self._smoke_tests:
-            self.run_test(
-                exe=join_path(self._smoke_tests_path, smoke_test[0]),
-                options=smoke_test[1],
-                status=[0],
-                installed=True,
-                skip_missing=True,
-                purpose=smoke_test[2],
-            )
+        self.run_example(join_path("cvode", "cuda", "cvAdvDiff_kry_cuda"), [], True)
 
-    def clean_smoke_tests(self):
-        if "~examples-install" in self.spec:
-            return
-
-        for smoke_test in self._smoke_tests:
-            work_dir = join_path(self._smoke_tests_path, os.path.dirname(smoke_test[0]))
-            with working_dir(work_dir):
-                self.run_test(exe="make", options=["clean"])
-
-    def test(self):
-        self.build_smoke_tests()
-        self.run_smoke_tests()
-        self.clean_smoke_tests()
-        return
+    def test_nvector_hip(self):
+        """build and run ROCM N_Vector"""
+        if "+rocm" not in self.spec:
+            raise SkipTest("Package must be installed with +rocm")
+
+        self.run_example(join_path("nvector", "hip", "test_nvector_hip"), ["10", "0", "0"], True)
+
+    def test_cvadvdiff_hip(self):
+        """build and run ROCM cvAdvDiff_kry"""
+        if "+rocm" not in self.spec or "+CVODE" not in self.spec:
+            raise SkipTest("Package must be installed with +rocm+CVODE")
+
+        self.run_example(join_path("cvode", "hip", "cvAdvDiff_kry_hip"), [], True)
+
+    def test_nvector_sycl(self):
+        """build and run SYCL N_Vector"""
+        if "+sycl" not in self.spec:
+            raise SkipTest("Package must be installed with +sycl")
+
+        self.run_example(
+            join_path("nvector", "sycl", "test_nvector_sycl"), ["10", "0", "0"], False
+        )
+
+    def test_sycl_cvode(self):
+        """build and run SYCL cvAdvDiff_kry"""
+        if "+sycl" not in self.spec or "+CVODE" not in self.spec:
+            raise SkipTest("Package must be installed with +sycl and +CVODE")
+
+        self.run_example(join_path("cvode", "sycl", "cvAdvDiff_kry_sycl"), [], True)
```

Comment:
Not for this PR necessarily, but would it be possible to decorate these with `@when` instead of having to have the conditional `SkipTest` error?

Reply:
Unless something has changed not involving https://github.com/spack/spack/pull/27789, that feature is not supported.  BUT I haven't tested it in a while.

Reply:
Ugh I forgot about that PR, sorry for the noise.

Yeah that would be a great addition later.